### PR TITLE
Switch olbase to use python:3.9.4 image instead of ubuntu:xenial

### PR DESCRIPTION
**Pull request by cdrini**
Experiment: pyenv is a good chunk of our build time, and a big chunk of the complexity of our container. We needed both python 2 and 3 installed to make the 2 to 3 migration easier; hence pyenv. Since future python migrations are very unlikely to be as complicated, we should just switch to hammering in one version of python. This reduces build times (since pyenv _builds_ python), and greatly reduces complexity: simpler Dockerfile, no more environment variables. It also means updating python is much simpler; change 3.9.4 to whatever, put on testing.openlibrary.org and test.

Note this does mean we're no longer based off ubuntu:xenial ; I can't figure out if python:3.9.4 is ubuntu based, but apt-get works, and creating/managing users works, so... 🤷‍♂️

### Technical
- The resulting image size is about the same (~2.04 GB ungzipped)

### Testing
- [x] Test local environment works
- [ ] Push up image to docker hub with `pybase` label
- [ ] Put it on testing
- [ ] Restart crons on ol-home0 to use the pybase image
- [ ] Update cron scripts to not do any python magic

### Stakeholders
@mekarpeles @dhruvmanila @cclauss

### CODE REVIEW

File: docker/Dockerfile.olbase [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:xenial
+FROM python:3.9.4
```
Comment (cclauss) [inline docker/Dockerfile.olbase:1]: Py3.9.6 is current  https://www.python.org/downloads

Now avail... Py3.10 release candidate 1... https://pythoninsider.blogspot.com/2021/08/python-3100rc1-is-available.html

**Review by cclauss — APPROVED** [review]
LGTM if tests pass.

**Comment by jimman2003** [comment]
It is using debian :) see tags at dockerhub

File: docker/Dockerfile.olbase (hunk 1 above)
Comment (cdrini) [reply docker/Dockerfile.olbase:1]: One step at a time :D

**Comment by cdrini** [comment]
@jimman2003 Ah, ok; hopefully switching ubuntu -> debian won't cause any issues!

**Comment by cdrini** [comment]
@cclauss Not sure why the tests have stalled...

**Comment by jimman2003** [comment]
https://www.githubstatus.com/incidents/7p1nnvkgh96y

**Comment by jimman2003** [comment]
is smth blocking this?

**Comment by cclauss** [comment]
Maybe jump to 3.9.7?  https://docs.python.org/release/3.9.7/

**Pull request closed without merging**